command/agent/retry_join.go

```diff
@@ -82,6 +82,33 @@ func (r *retryJoiner) Validate(config *Config) error {
 		}
 	}
 
+	if config.Server != nil {
+		dur, err := time.ParseDuration(config.Server.RetryInterval)
```

Comment:
Just change the struct to parse directly into the time value to avoid this?

https://github.com/hashicorp/nomad/blob/c42a07b488c5cb22350ea988234485394a477ebc/command/agent/config.go#L300